eclass/zig-toolchain.eclass

```diff
@@ -0,0 +1,382 @@
+#
+# Can be overriden by user.  If not overriden, then set by
+# "zig-toolchain_setup".
+#
+# Example:
+# @CODE
+# # Autodetected by Zig:
+# native
+# # Machine running Linux x86_64 system, with glibc:
+# x86_64-linux-gnu
+# # Similar to above, but versions are passed explicitly:
+# x86_64-linux.6.1.12...6.6.16-gnu.2.38
+# # Machine running Linux PPC64 little-endian system, with musl
+# powerpc64le-linux-musl
+# @CODE
+#
+
+# @ECLASS_VARIABLE: ZIG_CPU
+# @DESCRIPTION:
+# Zig target CPU and features to use.  Has the following format:
+# family_name(\+enable_feature|\-disable_feature)*
+# Can be passed as:
+# * "-mcpu " option in "zig test" or "zig build-exe/lib/obj",
+# * "-Dcpu=" option in "zig build"
+#   (if project uses "std.Build.standardTargetOptions").
+#
+# Can be overriden by user.  If not overriden, then set by
+# "zig-toolchain_setup".
+#
+# Example:
+# @CODE
+# # Autodetected by Zig:
+# native
+# # AMD Zen 2 processor
+# znver2
+# # x86_64 processor, X87 support enabled, SSE2 support disabled
+# x86_64+x87-sse2
+# @CODE
+
+# @ECLASS_VARIABLE: ZIG_EXE
+# @DESCRIPTION:
+# Absolute path to the used Zig executable.
+#
+# Please note that when passing one flag several times with different
+# values:
+# * (only "zig build") in "-Dbar=false -Dbar" form:
+#   errors due to conflict of flags,
+# * (only "zig build") in "-Dbar=false -Dbar=true" form:
+#   "bar" becomes a list, which is likely not what you want,
+# * in "-fbar -fno-bar" form:
+#   latest value overwrites values before.
+# Example above shows only boolean option, but it is same with other
+# types of options (enums, "std.zig.BuildId", "std.SemanticVersion",
+# integers, strings, etc.).
+#
+# Can be overriden by user.  If not overriden, then set by
+# "zig-toolchain_setup".
+
+# @ECLASS_VARIABLE: ZIG_VER
+# @OUTPUT_VARIABLE
+# @DESCRIPTION:
+# Zig version found during "zig-toolchain_find_installation",
+# as reported in dev-lang/zig-${PV} PV part.
+#
+# Example:
+# @CODE
+# 0.13.0
+# @CODE
+
+# @FUNCTION: zig-toolchain_get_target
+# @USAGE: <C-style target triple>
+# @DESCRIPTION:
+# Translates C-style target triple (like CHOST or CBUILD)
+# to Zig-style target triple.  Some information (like ARM features)
+# is handled by "zig-toolchain_get_cpu".  Mostly used during
+# cross-compiling if user does not set ZIG_TARGET variable.
+#
+# See ZIG_TARGET description for more information.
+zig-toolchain_get_target() {
+	if [[ ${#} -ne 1 ]]; then
+		die "${FUNCNAME[0]}: expected 1 arguments, got ${#}"
+	fi
+	local c_target=${1}
+	local c_target_prefix=${c_target%%-*}
+	local c_target_suffix=${c_target##*-}
+
+	local arch os abi
+
+	case ${c_target_prefix} in
+		i?86)	arch=x86;;
+		arm64)	arch=aarch64;;
+		arm*)	arch=arm;;
+		*)		arch=${c_target_prefix};;
+	esac
+
+	case ${c_target} in
+		*linux*)	os=linux;;
+		*apple*)	os=macos;;
+	esac
+
+	case ${c_target_suffix} in
+		solaris*)	os=solaris abi=none;;
+		darwin*)	abi=none;;
+		*)			abi=${c_target_suffix};;
+	esac
```

Comment:
Looks like some poor man's alignment attempt that went all messed up.

Reply:
It's all aligned well in my browser, and at mailing list they asked to format it this way, maybe it's on your side

Reply:
There should only be one tab, not three.

Reply:
The point is, alignment never works on two systems alike, so don't do it.

Reply:
a9b50921a624...ad1c44530f74